nautobot_design_builder/jobs.py

```diff
@@ -17,23 +17,39 @@ class DeploymentDecommissioning(Job):
         description="Design Deployments to decommission.",
     )
 
+    delete = BooleanVar(
+        description="Actually delete the objects, not just their link to the design deployment.",
+        default=True,
+    )
+
     class Meta:  # pylint: disable=too-few-public-methods
         """Meta class."""
 
         name = "Decommission Design Deployments"
         description = """Job to decommission one or many Design Deployments from Nautobot."""
 
-    def run(self, deployments):  # pylint:disable=arguments-differ
+    def run(self, deployments, delete):  # pylint:disable=arguments-differ
         """Execute Decommissioning job."""
         self.logger.info(
             "Starting decommissioning of design deployments: %s",
             ", ".join([instance.name for instance in deployments]),
         )
 
         for deployment in deployments:
-            self.logger.info("Working on resetting objects for this Design Instance...", extra={"object": deployment})
-            deployment.decommission(local_logger=self.logger)
-            self.logger.info("%s has been successfully decommissioned from Nautobot.", deployment)
+            if delete:
+                message = "Working on deleting objects for this Design Deployment."
+            else:
+                message = "Working on unlinking objects from this Design Deployment."
+            self.logger.info(message, extra={"object": deployment})
+
+            deployment.decommission(local_logger=self.logger, delete=delete)
+
+            if delete:
+                message = f"{deployment} has been successfully decommissioned from Nautobot."
+            else:
+                message = f"Objects have been successfully unlinked from {deployment}."
+
+            self.logger.info(message)
```

Comment:
This is really an antipattern. Python loggers are optimized to use *args and format strings and only expand the string if the correct log level is selected. I suggest the following:

```suggestion
            if delete:
                self.logger.info("%s has been successfully decommissioned from Nautobot.", deployment)
            else:
                self.logger.info("Objects have been successfully unlinked from %s", deployment)
```